src/Libraries/RevitNodesUI/GenericClasses.cs

```diff
@@ -0,0 +1,175 @@
+    /// It uses a filtered element collector filtering by class.
+    /// </summary>
+    public abstract class CustomRevitElementDropDown : RevitDropDownBase
+    {
+        /// <summary>
+        /// Generic Revit Element Class Dropdown Node
+        /// </summary>
+        /// <param name="name">Name of the Node</param>
+        /// <param name="elementType">Type of Revit Element to display</param>
+        public CustomRevitElementDropDown(string name, Type elementType) : base(name) { this.ElementType = elementType; PopulateItems(); }
+
+        /// <summary>
+        /// Constant message for missing Types
+        /// </summary>
+        private const string noTypes = "No Types available.";
+
+        /// <summary>
+        /// Type of Element
+        /// </summary>
+        public Type ElementType;
+
+        protected override CoreNodeModels.DSDropDownBase.SelectionState PopulateItemsCore(string currentSelection)
+        {
+            PopulateItems();
+            return SelectionState.Done;
+        }
+        /// <summary>
+        /// Populate the Dropdown menu
+        /// </summary>
+        public void PopulateItems()
+        {
+            if (this.ElementType != null)
+            {
+                // Clear the Items
+                Items.Clear();
+
+                // Set up a new element collector using the Type field
+                var fec = new RVT.FilteredElementCollector(DocumentManager.Instance.CurrentDBDocument).OfClass(ElementType);
+
+                // If there is nothing in the collector add the missing Type message to the Dropdown menu.
+                if (fec.ToElements().Count == 0)
+                {
+                    Items.Add(new CoreNodeModels.DynamoDropDownItem(noTypes, null));
+                    SelectedIndex = 0;
+                    return;
+                }
+
+                // if the elementtype is a RebarHookType add an initial "None" value which does not come from the collector
+                if (this.ElementType.FullName == "Autodesk.Revit.DB.Structure.RebarHookType") Items.Add(new CoreNodeModels.DynamoDropDownItem("None", null));
+
+                // Walk through all elements in the collector and add them to the dropdown
+                foreach (var ft in fec.ToElements())
+                {
+                    Items.Add(new CoreNodeModels.DynamoDropDownItem(ft.Name, ft));
+                }
+
+                Items = Items.OrderBy(x => x.Name).ToObservableCollection();
+            }
+        }
+
+        /// <summary>
+        /// Cast the selected element to a dynamo node
+        /// </summary>
+        public override IEnumerable<AssociativeNode> BuildOutputAst(List<AssociativeNode> inputAstNodes)
+        {
+            // If there are no elements in the dropdown or the selected Index is invalid return a Null node.
+            if (Items.Count == 0 || 
+            Items[0].Name == noTypes ||
+            SelectedIndex == -1 || Items[SelectedIndex].Name == "None")
```

Comment:
I see the "none" type now is used for this comparison... so it shouldn't goto the .resx unless exposed to the user, instead just pull it out as a constant. any reason not to use `null`?